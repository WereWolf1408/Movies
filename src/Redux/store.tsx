import { configureStore } from '@reduxjs/toolkit';
import { useDispatch } from 'react-redux';
import movies from './reducers/moviesReducer';
import modalWindows from './reducers/modalWindowsReducer';
import additionalStore from './reducers/additionalReducer';

export const store = configureStore({
  reducer: {
    movies,
    additionalStore,
    modalWindows,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export const AppDispatch = () => useDispatch<typeof store.dispatch>();
