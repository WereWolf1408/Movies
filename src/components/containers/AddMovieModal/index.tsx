import React from 'react';
import ModalWindow from '@common/ModalWindow';
import Form from '@common/Form';
import { formDataReceived } from '@utils/interfaces';

const CLASSES = {
  NETFLIX_APP_FORM_MODAL_WINDOW: 'netflix-app__form-modal-window',
};

interface AddMoviewModalProps {
  (props: {}): JSX.Element;
}

const AddMoviewModal: AddMoviewModalProps = ({}) => {
  return (
    <ModalWindow title={'add movie'}>
      <Form />
    </ModalWindow>
  );
};

export default AddMoviewModal;
