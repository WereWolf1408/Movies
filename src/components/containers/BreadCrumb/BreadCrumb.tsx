import classNames from 'classnames';
import React, { useState } from 'react';
import {Dropdown} from '../../common/Dropdown';
import { sortOptions } from '@utils/utils';

import './BreadCrumb.less';

const genres = ['all', 'Documentary', 'comedy', 'horror', 'crime'];

const CLASSES = {
  NETFLIX_APP_BREADCRUMB: 'netflix-app__breadcrumb',
  NETFLIX_APP_BREADCRUMB_GENRE: 'netflix-app__breadcrumb-genre',
  NETFLIX_APP_BREADCRUMB_SORT: 'netflix-app__breadcrumb-sort',
  NETFLIX_APP_BREADCRUMB_ITEM: 'netflix-app__breadcrumb-item',
  NETFLIX_APP_BREADCRUMB_LABEL: 'netflix-app__breadcrumb-label',
  ACTIVE: 'active',
};

type BreadCrumbItemProps = (props: {
  genre: string;
  clickHandler: (activeId: number) => void;
  activeId: number;
  isActive: boolean;
}) => JSX.Element;

export const BreadCrumbItem: BreadCrumbItemProps = ({
  genre,
  clickHandler,
  activeId,
  isActive,
}) => (
  <span
    className={classNames(CLASSES.NETFLIX_APP_BREADCRUMB_ITEM, {
      [CLASSES.ACTIVE]: isActive,
    })}
    onClick={() => clickHandler(activeId)}
  >
    {genre}
  </span>
);

export const BreadCrumb = () => {
  const [activeItem, setActiveItem] = useState(0);

  const clickHandler = (activeId: number) => {
    setActiveItem(activeId);
  };

  const renderGenreList = () =>
    genres.map((genre, index) => (
      <BreadCrumbItem
        key={index}
        genre={genre}
        clickHandler={clickHandler}
        activeId={index}
        isActive={activeItem === index}
      />
    ));

  return (
    <section className={CLASSES.NETFLIX_APP_BREADCRUMB}>
      <div className={CLASSES.NETFLIX_APP_BREADCRUMB_GENRE}>{renderGenreList()}</div>
      <div className={CLASSES.NETFLIX_APP_BREADCRUMB_SORT}>
        <span className={CLASSES.NETFLIX_APP_BREADCRUMB_LABEL}>Sort by</span>
        <Dropdown options={sortOptions} dropdownType={'simple'} />
      </div>
    </section>
  );
};