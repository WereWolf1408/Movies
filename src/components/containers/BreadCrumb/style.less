@import '@styles/colors.less';

.netflix-app__breadcrumb {
  display: flex;
  padding: 2rem 0 1.9rem;
  border-bottom: 2px solid @color-cod-gray;
  color: @color-white;

  &-genre {
    flex-grow: 1;
  }

  &-item,
  &-label {
    font-size: 1.6rem;
    text-transform: uppercase;
    font-weight: 600;
    cursor: pointer;
  }

  &-item {
    position: relative;

    &:not(:first-child) {
      margin-left: 3rem;
    }

    &:after {
      content: "";
      position: absolute;
      display: none;
      height: .4rem;
      width: 100%;
      top: 3.4rem;
      left: 0;
      background: red;
    }

    &.active {
      &:after {
        display: block;
      }
    }
  }

  &-label {
    margin-right: 3rem;
  }

  &-sort {
    display: flex;

    .netflix-app__breadcrumb-label {
      opacity: .6;
    }
  }
}
