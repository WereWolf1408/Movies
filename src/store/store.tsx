import { configureStore } from '@reduxjs/toolkit';
import { useDispatch } from 'react-redux';
import { rootReducer } from './rootReducer';
import { additionalReducer } from './additionalReducer';

export const store = configureStore({
  reducer: {
    mainStore: rootReducer,
    additionalStore: additionalReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export const AppDispatch = () => useDispatch<typeof store.dispatch>();
