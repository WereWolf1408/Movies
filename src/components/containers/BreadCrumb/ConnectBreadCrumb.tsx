import React from 'react';
import { connect } from 'react-redux';
import { MapDispatchProps } from '../../../utils/interfaces';
import { BreadCrumb } from './BreadCrumb';
import { applySort } from '../../../store/actions';

const mapDispatchToProps: MapDispatchProps<
  {},
  { applySortCallback: (value: string) => void }
> = (dispatch) => {
  return {
    applySortCallback: (sortValue: string) => dispatch(applySort(sortValue)),
  };
};

export default connect(null, mapDispatchToProps)(BreadCrumb);
