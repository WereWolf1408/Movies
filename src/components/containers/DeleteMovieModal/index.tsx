import React from 'react';
import Button from '../../common/Button';
import ModalWindow from '../../common/ModalWindow';

import './style.less';

const CLASSES = {
  NETFLIX_APP_DELETE_MODAL: 'netflix-app__delete-movie',
  NETFLIX_APP_DELETE_MODAL_TEXT: 'netflix-app__delete-movie-text',
  NETFLIX_APP_DELETE_MODAL_BUTTON: 'netflix-app__delete-movie-button',
};

const DeleteMovieModal = () => {
  return (
    <ModalWindow title={'delete movie'}>
      <div className={CLASSES.NETFLIX_APP_DELETE_MODAL}>
        <span className={CLASSES.NETFLIX_APP_DELETE_MODAL_TEXT}>
          Are you sure you want to delete this movie?
        </span>
        <Button
          classes={CLASSES.NETFLIX_APP_DELETE_MODAL_BUTTON}
          clickHandler={() => {}}
          text={'confirm'}
        />
      </div>
    </ModalWindow>
  );
};

export default DeleteMovieModal;
