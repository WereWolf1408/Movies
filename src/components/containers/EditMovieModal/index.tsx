import React, { useContext } from 'react';
import { NetflixAppContext } from '../../../Context';
import Button from '../../common/Button';
import Form from '../../common/Form';
import ModalWindow from '../../common/ModalWindow';

import './style.less';

const CLASSES = {
  NETFLIX_APP_EDIT_MODAL: 'netflix-app__edit-movie',
};

const DeleteMovieModal = () => {
  const { movieDetail } = useContext(NetflixAppContext);
  return (
    <ModalWindow title={'Edit Movie'}>
      <div className={CLASSES.NETFLIX_APP_EDIT_MODAL}>
        <Form titleValue={movieDetail.props.title} genreValue={movieDetail.props.genre} />
      </div>
    </ModalWindow>
  );
};

export default DeleteMovieModal;
