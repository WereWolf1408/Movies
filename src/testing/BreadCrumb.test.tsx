import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { BreadCrumb } from '../components/containers/BreadCrumb';
import { store } from '../store/store';

describe('BreadCrumb Component', () => {
  let mockStore: any;
  beforeEach(() => {
    mockStore = store;
  });

  it('---> shapshot test:', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    expect(asFragment).toMatchSnapshot();
  });

  it(`---> user interaction test`, async () => {
    const { getByText } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    const breadCrumbItem = getByText(/Documentary/i);
    expect(breadCrumbItem).toBeInTheDocument();
    expect(breadCrumbItem).not.toHaveClass('active');
    userEvent.click(breadCrumbItem);
    expect(breadCrumbItem).toHaveClass('active');
  });

  it(`---> BreadCrumb test dropdown`, () => {
    const { getByText } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    const dropdown = getByText('select value');
    expect(dropdown).toBeInTheDocument();
    userEvent.click(dropdown);
    expect(getByText('title')).toBeInTheDocument();
  });
});
