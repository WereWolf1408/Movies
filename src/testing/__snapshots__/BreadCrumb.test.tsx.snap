// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Component ---> shapshot test: 1`] = `[Function]`;
